src/main/java/org/mockito/listeners/VerificationListener.java

```diff
@@ -0,0 +1,20 @@
+package org.mockito.listeners;
+
+import org.mockito.internal.verification.api.VerificationData;
+import org.mockito.verification.VerificationMode;
+
+/**
+ * This listener can be notified of verify invocations on a mock.
+ *
+ * For this to happen, it must be registered using {@link org.mockito.internal.progress.MockingProgress#addListener(MockitoListener)}.
+ */
+public interface VerificationListener extends MockitoListener {
+    /**
+     * Called when a verification happened.
+     *
+     * @param mock the mock that verify was called with.
+     * @param mode the mode that the mock will be verified with.
+     * @param verificationData the data of this verification.
+     */
+    void onVerification(Object mock, VerificationMode mode, VerificationData verificationData);
```

Comment:
Makes a lot of sense, done.

My use cases is however not interested in the case that verify does throw. 
Would it however make any sense to add support for it?